UnitTests/UnitTests.UWP/Helpers/SystemTextJsonSerializer.cs

```diff
@@ -13,8 +13,8 @@ namespace UnitTests.UWP.Helpers
     /// </summary>
     internal class SystemTextJsonSerializer : IObjectSerializer
     {
-        public T Deserialize<T>(object value) => JsonSerializer.Deserialize<T>(value as string);
+        public T Deserialize<T>(string value) => JsonSerializer.Deserialize<T>(value);
 
-        public object Serialize<T>(T value) => JsonSerializer.Serialize(value);
+        public string Serialize<T>(T value) => JsonSerializer.Serialize(value);
```

Comment:
Note for the future: this architecture makes it impossible to leverage the new source generator support for the `System.Text.Json` APIs on .NET 6, and it will just force the original reflection-based path. We should eventually reconsider this as well when we get to plan a better revamp of this whole group of APIs 🙂

Reply:
I agree, this should be thought out more in the future. But in the meantime I'm all ears on suggestions for a happy medium 😊

Reply:
Good point, after 7.1, we should figure out a plan for how all this works in the .NET space and able to leverage all the JSON Source Generator goodness as that looks amazing.